### ISSUE TITLE
NormalizeSynthesizedMemberOrdinals folds two overloads' local functions into one operand, masking a real difference

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#3645]
`NormalizeSynthesizedMemberOrdinals` (landed in #3505 for #3503) masks a real difference between the local functions of two overloads. Its own remarks name this case as a known limitation, then conclude with a claim that is false for it:

> Known limitation, deliberately not addressed: overloads share a containing-method name and are told apart only by `N`, so this option conflates the closures of two overloads with the same lambda index. [...] **Each of these costs a false positive, never a masked difference.**

The first item costs a **masked difference**, not a false positive. Conflating two overloads' closures relates two members that genuinely differ, which is a false `Exact` — the one direction this normalization was designed never to take.

### Measured, on real Roslyn output

```csharp
public static class C
{
    public static int M(int x)    { return a(); static int a() => 1; }
    public static int M(string x) { return a(); static int a() => 2; }
}
```

Roslyn emits `<M>g__a|0_0` and `<M>g__a|1_0` — same declaring type, same signature `int32()`, genuinely different code (one returns 1, the other 2). Comparing the two overload bodies with `IlBodyDiff.Compare(reader, b0, reader, b1, normalization)`:

| normalization | outcome |
| --- | --- |
| `None` | `OperandDiff` — `call int32 [Ovl]C::<M>g__a\|0_0()` vs `call int32 [Ovl]C::<M>g__a\|1_0()` |
| `NormalizeSynthesizedMemberOrdinals` | **`Exact`** |

Both names normalize to `<M>g__a|#_0`, and nothing else in the operand distinguishes them.

### Why this matters beyond the doc comment

`FidelityCheck.cs:52` puts this flag in the normalization set the compile-back fidelity oracle runs with. So the oracle can report `Exact` for a decompilation that binds a call to the *wrong overload's* local function — which is exactly the kind of misattribution the oracle exists to detect. The failure is invisible: it produces a success-shaped result, not a diagnostic.

No gate covers it. `IlBodyDiffNormalizationTests` has no test mentioning overloads; the sole match for "overload" in that file is unrelated prose. The limitation was documented and then not pinned, so nothing fails today and nothing will fail if it widens.

### Root cause is the design, not the parser

The rewrite is **per-side**: each name is normalized with no knowledge of the other side, so it cannot ask whether the normalized key is unique. Uniqueness is a two-sided property, so no amount of tightening the name grammar can express it — the parser is already correct about *which* names it recognizes; the problem is that recognizing them is not sufficient grounds to relate them.

### Confirmed fix direction

The correspondence in #3580 (`NormalizeCompilerGeneratedOrdinals`) refuses precisely this case: it indexes both sides, and folds only where the ordinal-free key is one-to-one **on both sides**. `<M>g__a|#_0` is ambiguous, so the fold is declined. Same probe, same assembly, same API:

| normalization | outcome |
| --- | --- |
| `NormalizeCompilerGeneratedOrdinals` (#3580) | `OperandDiff` — correctly still a difference |

#3580 already carries `RealOverloadShape_IsRefusedAsAmbiguous` as an artifact canary pinning this exact Roslyn shape.

Note the two options are not ordered by scope: #3505 covers `b__` and `<>9__`, which #3580 does not; #3580 covers `d__`, which #3505 declines. Resolving the overlap is a separate decision from fixing the soundness hole, and I'll raise it on #3580.

### Repro

`/home/rich/probe-ovl.cs` against `main` @ `ec7b254ef`; fixture source above.
